tools/proofers/preview.js

```diff
@@ -26,16 +26,26 @@ function processExMath(text, textFunction, allowMath) {
     }
 }
 
-$(function () {
-    function removeComments(textLine) {
-        return textLine.replace(/\[\*\*[^\]]*\]/g, '');
-    }
-
-    // true if txtLine contains anything except only comments or spaces
-    function nonComment(textLine) {
-        return (/\S/.test(removeComments(textLine)));
+// find index of next unmatched ] return 0 if none found
+function findClose(txt, index) {
+    var result;
+    var nestLevel = 0;
+    var re = /\[|\]/g;  // [ or ]
```

Comment:
Not sure if it matters, but I'd put the regex out of the function, so it won't be created every function invocation. Compiling regex can be expensive and I've seen this trick cause an app go from 8 min to instant.

Reply:
Interesting. Done as you suggest.